### PR TITLE
SDIT-886 Add BatchSize to improve number of sql calls made

### PR DESCRIPTION
**Pull request by karenmillermoj**
One of the biggest sql queries (366 sql calls) is reduced to 110 using the batchsize annotations.  Further work to follow with inclusion of entity graph

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderBooking.java [hunk 1]
```diff
@@ -211,22 +213,26 @@ public class OffenderBooking extends AuditableEntity {
...
     @OneToMany(mappedBy = "offenderBooking", cascade = CascadeType.ALL)
     @Default
     @Exclude
+    @BatchSize(size = 1000)
     private List<SentenceTerm> terms = new ArrayList<>();
 
     @OneToMany(mappedBy = "offenderBooking", cascade = CascadeType.ALL)
     @Default
     @Exclude
-    private Set<OffenderSentence> sentences = new HashSet<>();
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderBooking.java:229]: No longer needs to be a Set (was from an earlier commit)

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/SentenceTermRepository.kt [hunk 2]
```diff
@@ -5,9 +5,8 @@ import org.springframework.data.jpa.repository.EntityGraph.EntityGraphType
 import org.springframework.data.repository.CrudRepository
 import uk.gov.justice.hmpps.prison.repository.jpa.model.SentenceTerm
 import uk.gov.justice.hmpps.prison.repository.jpa.model.SentenceTerm.PK
-import java.util.List
 
 public interface SentenceTermRepository : CrudRepository<SentenceTerm, PK> {
   @EntityGraph(type = EntityGraphType.FETCH, value = "sentence-term-with-offender-sentence")
-  fun findByOffenderBookingBookingId(bookingId: Long): List<SentenceTerm>
+  fun findByOffenderBookingBookingId(bookingId: Long): MutableList<SentenceTerm>
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/SentenceTermRepository.kt:11]: intelliJ showed warning for java.util.List

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderSentence.java [hunk 3]
```diff
@@ -103,7 +104,8 @@ public static class PK implements Serializable {
         @JoinColumn(name = "OFFENDER_BOOK_ID", referencedColumnName = "OFFENDER_BOOK_ID"),
         @JoinColumn(name = "SENTENCE_SEQ", referencedColumnName = "SENTENCE_SEQ")
     })
-    private Set<SentenceTerm> terms;
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderSentence.java:106]: Revert back to list

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**